primitives/traits/src/alias.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+#[doc(hidden)]
+pub mod __reexports {
+    pub use {
+        frame_support::{CloneNoBound, EqNoBound, PartialEqNoBound, RuntimeDebugNoBound},
+        scale_info::TypeInfo,
+        serde::{de::DeserializeOwned, Deserialize, Serialize},
+        sp_core::{Decode, Encode, RuntimeDebug},
+    };
+}
+
+pub use macro_rules_attribute::apply;
+
+#[macro_export]
+macro_rules! derive_storage_traits {
+    ( $( $tt:tt )* ) => {
+        #[derive(
+            $crate::alias::__reexports::RuntimeDebug,
+            ::core::cmp::PartialEq,
+            ::core::cmp::Eq,
+            ::core::clone::Clone,
+            $crate::alias::__reexports::Encode,
+            $crate::alias::__reexports::Decode,
+            $crate::alias::__reexports::TypeInfo,
+            $crate::alias::__reexports::Serialize,
+            $crate::alias::__reexports::Deserialize,
```

Comment:
Generally we don't need serde traits for storage, it's only needed if using a type in genesis, or in rpc. So I would rather see 2 derives above the type definition, one for storage traits and one for serde. And the serde alias is not that useful as its only 2 traits, it will probably look better if it's explicit.